scripts/monitoring/auto_scale_workers.py

```diff
@@ -33,101 +45,100 @@ def get_challenges():
     return response
 
 
-def get_sqs_queue_by_name(queue_name):
-    """
-    Returns:
-        Returns the SQS Queue object
-    """
-
-    sqs = boto3.resource(
-        "sqs",
-        region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
-        aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
-        aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
-    )
-    if queue_name == "":
-        raise ValueError("Queue Name cannot be empty")
+def get_queue_length(queue_name):
+    try:
+        num_processed_submissions = int(
+            prom.custom_query(
+                f"num_processed_submissions{{queue_name='{queue_name}'}}"
+            )[0]["value"][1]
+        )
+    except Exception as _e:  # noqa: F841
+        # If we are unable to fetch num_processes_submissions, we assume they are zero
+        num_processed_submissions = 0
 
     try:
-        queue = sqs.get_queue_by_name(QueueName=queue_name)
-    except botocore.exceptions.ClientError as ex:
-        if (
-            ex.response["Error"]["Code"]
-            != "AWS.SimpleQueueService.NonExistentQueue"
-        ):
-            raise Exception("Cannot get queue: {}".format(queue_name))
-    return queue
-
-
-def get_queue_length(queue):
-    # TODO: Check if we should use approximate number of messages
-    # https://github.com/boto/boto3/issues/599
-    # Ref: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/sqs.html#using-an-existing-queue
-    # Ref: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Queue.attributes
-    return (
-        queue.attributes["ApproximateNumberOfMessages"]
-        + queue.attributes["ApproximateNumberOfMessagesNotVisible"]
-    )
+        num_submissions_in_queue = int(
+            prom.custom_query(
+                f"num_submissions_in_queue{{queue_name='{queue_name}'}}"
+            )[0]["value"][1]
+        )
+    except Exception as _e:  # noqa: F841
```

Comment:
We are not using `_e` anywhere so just change to `except Exception`

Reply:
Resolved.